README.md

```diff
@@ -5,11 +5,11 @@
 
 Geode Loopless (GeodeL) is a Domain Specific Language that aims at facilitating the writing and analysis of Primitive Recursive Functions and Sets.
 
-This DSL is embedded in Scala, and aims to use it's strong type system to enforce the arity and type of the PrimRec functions and sets, thus ensuring any function or set is really primitive recursive.
+This DSL is embedded in Scala, and aims to use its strong type system to enforce the arity and type of the `PrimRec` functions and sets, thus ensuring any function or set is really primitive recursive.
 
-Primitive recursive functions are all the functions created by combining the successor function and constants with projection, recursion and composition.
+Primitive recursive functions are all the functions that can be created by combining the successor function on naturals and constants with projection, recursion, and composition.
```

Comment:
```suggestion
Primitive recursive functions are all the functions on naturals that can be created by combining the successor function and constants with projection, recursion, and composition.
```
Or maybe
```suggestion
Primitive recursive functions are all the functions on tuples of naturals that can be created by combining the successor function and constants with projection, recursion, and composition.
```

Reply:
Also what do you think of this ?
```suggestion
Primitive recursive functions are all the functions that can be created by combining constants, projections, and the successor function through recursion and composition.
```

Reply:
```suggestion
Primitive recursive functions are all the functions on naturals that can be created by combining constants, projections, and the successor function through recursion and composition.
```

I like the combination of changes 1 and 3 here.

Reply:
Yes I agree, these were intended to be combined !